contracts/transmuter/facets/Swapper.sol

```diff
@@ -443,12 +447,13 @@ contract Swapper is ISwapper {
             _computeFee(quoteType, amountStable, v.isMint ? collatInfo.yFeeMint[n - 1] : collatInfo.yFeeBurn[n - 1]);
     }
 
-    /// @notice Checks whether a managed collateral asset still has enough collateral available to process
-    /// a transfer
-    function _checkAmounts(Collateral storage collatInfo, uint256 amountOut) internal view {
+    /// @notice Checks whether there is still enough of the collateral to process the transfer
+    function _checkAmounts(address collateral, Collateral storage collatInfo, uint256 amountOut) internal view {
         // Checking if enough is available for collateral assets that involve manager addresses
-        if (collatInfo.isManaged > 0 && LibManager.maxAvailable(collatInfo.managerData.config) < amountOut)
-            revert InvalidSwap();
+        if (
+            (collatInfo.isManaged > 0 && LibManager.maxAvailable(collatInfo.managerData.config) < amountOut) ||
+            (IERC20(collateral).balanceOf(address(this)) < amountOut)
```

Comment:
is it really identical to `collatInfo.isManaged == 0 && IERC20(collateral).balanceOf(address(this)) < amountOut`